zcash_primitives/src/transaction/components/sapling.rs

```diff
@@ -56,11 +56,14 @@ impl Authorization for Authorized {
     type AuthSig = redjubjub::Signature;
 }
 
+/// A map from one bundle authorization to another.
+///
+/// For use with [`Bundle::map_authorization`].
 pub trait MapAuth<A: Authorization, B: Authorization> {
-    fn map_spend_proof(&self, p: A::SpendProof) -> B::SpendProof;
-    fn map_output_proof(&self, p: A::OutputProof) -> B::OutputProof;
-    fn map_auth_sig(&self, s: A::AuthSig) -> B::AuthSig;
-    fn map_authorization(&self, a: A) -> B;
+    fn map_spend_proof(&mut self, p: A::SpendProof) -> B::SpendProof;
+    fn map_output_proof(&mut self, p: A::OutputProof) -> B::OutputProof;
+    fn map_auth_sig(&mut self, s: A::AuthSig) -> B::AuthSig;
+    fn map_authorization(&mut self, a: A) -> B;
```

Comment:
`map` with a `&mut self` reference seems increasingly misnamed, and the commit text doesn't explain the rationale for this change, so I'm assuming I'll be able to figure it out from later commits.

Reply:
I gave an example of why I needed this in a comment on the PR.